args.zig

```diff
@@ -98,7 +97,22 @@ fn parseInternal(comptime Generic: type, comptime MaybeVerb: ?type, args_iterato
 
     var last_error: ?anyerror = null;
 
-    while (args_iterator.next()) |item| {
+    // Create map for required arguments
+    var required_map = std.StringHashMap(bool).init(allocator);
```

Comment:
Can you replace the `StringHashMap` with a [`std.EnumSet`](https://github.com/ziglang/zig/blob/74ed7c1f0998e9dd89aa3f3480fff845afd6b422/lib/std/enums.zig#L210)? As we know all possible fields, we can use [`std.meta.FieldEnum`](https://github.com/ziglang/zig/blob/74ed7c1f0998e9dd89aa3f3480fff845afd6b422/lib/std/meta.zig#L607) instead of runtime allocation

Reply:
> How do you get a proper enum to insert in this set from a field?

```zig
const Fields = std.meta.FieldEnum(Type);

const field_member = @field(Fields, field.name);

set.insert(field_member);
```

> Also, how would this work with the required items for the tagged enum of the Verb set, do we need a set per verb?

Yes, but that would not hurt much, as we have to process the verb separately anyways. As we only process more errors when mandatory verb parameters are missing, this should not be much of a problem